server/models/user.js

```diff
@@ -0,0 +1,78 @@
+
+/**
+ * @description define User database model
+ * @param {Object} sequelize - sequelize object
+ * @param {Object} DataTypes - data types
+ * @returns {Object} - instance of User database model
+ */
+module.exports = (sequelize, DataTypes) => {
+  const User = sequelize.define('User', {
+    firstName: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      validate: {
+        notEmpty: true,
+        isAlpha: true
+      }
+    },
+    lastName: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      validate: {
+        notEmpty: true,
+        isAlpha: true
+      }
+    },
+    username: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      unique: true,
+      validate: {
+        notEmpty: true,
+        isAlphanumeric: true
+      }
+    },
+    email: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      unique: true,
+      validate: {
+        isEmail: {
+          args: true,
+          msg: 'email address entered not valid'
+        }
+      }
+    },
+    password: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      validate: {
+        notEmpty: true,
+        min: 6
+      }
+    },
+    blocked: {
+      type: DataTypes.BOOLEAN,
+      allowNull: false,
+      defaultValue: false
+    },
+    roleId: {
+      type: DataTypes.INTEGER,
+      defaultValue: 3,
+      validate: {
+        min: 2,
+        max: 5
+      }
+    }
+  });
+  User.associate = (models) => {
+    User.hasMany(models.Document, {
+      foreignKey: 'userId'
+    });
+    User.belongsTo(models.Role, {
+      foreignKey: 'roleId',
+      onDelete: 'CASCADE',
```

Comment:
Deleting several users (for example) when a role is deleted may not be the ideal thing to do in this case, you may to consider less devastating alternatives.